cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -140,6 +142,14 @@ data QueryInMode mode result where
     :: ConsensusMode mode
     -> QueryInMode mode ChainPoint
 
+instance NtcVersionOf (QueryInMode mode result) where
```

Comment:
This is exactly what I had in mind except as a function rather than a type class :smile:.